### PR TITLE
Add empty endpoints for rules

### PR DESCRIPTION
**Pull request by jillcary**
Added endpoints for CRUD rule operations with empty responses.

Documentation can be found here:
https://github.com/Azure/device-telemetry-java/wiki/%5BAPI-Specifications%5D-Rules

---
This change is [Reviewable](https://reviewable.io/reviews/azure/device-telemetry-java/26)

### CODE REVIEW

File: app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/RulesController.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     * Create a new rule with the specified id. Id must be unique.
+     *
+     * @return new rule created.
+     */
+    public Result create() { return ok(toJson("TODO")); }
+
+    /**
+     * @return updated rule.
+     */
+    public Result patch(String id) {
```
Comment (dluc) [inline app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/RulesController.java]: pls rename to post()

**Comment by jillcary** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/RulesController.java, line 40 at r2](https://reviewable.io:443/reviews/azure/device-telemetry-java/26#-Ko8gTgm4tl7PvBqlM3F-r2-40:-KoIz4ZO6tOiSG1d6r8w:b-896fix) ([raw file](https://github.com/azure/device-telemetry-java/blob/75d865b449165d1ac92b1032246b3bddbcea6e86/app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/RulesController.java#L40)):*
<details><summary><i>Previously, dluc (Devis Lucato) wrote…</i></summary><blockquote>

pls rename to post()
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/azure/device-telemetry-java/26)*

File: app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/RulesController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    public Result get(String id) {
+        return ok(toJson("TODO"));
+    }
+
+    /**
+     * Create a new rule with the specified id. Id must be unique.
+     *
+     * @return new rule created.
+     */
+    public Result create() { return ok(toJson("TODO")); }
```
Comment (dluc) [inline app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/RulesController.java]: for consistency, we should name the method `post` e.g. see https://github.com/Azure/iothub-manager-java/blob/master/app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/controllers/DevicesController.java

**Comment by jillcary** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/RulesController.java, line 35 at r3](https://reviewable.io:443/reviews/azure/device-telemetry-java/26#-Ko8gTgm4tl7PvBqlM3F-r3-35:-KohkE7OWvN645IaflNn:b-896fix) ([raw file](https://github.com/azure/device-telemetry-java/blob/79655dbb73e16b8a959530d6e601621d180fdebd/app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/RulesController.java#L35)):*
<details><summary><i>Previously, dluc (Devis Lucato) wrote…</i></summary><blockquote>

for consistency, we should name the method `post` e.g. see https://github.com/Azure/iothub-manager-java/blob/master/app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/controllers/DevicesController.java
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/azure/device-telemetry-java/26)*

File: conf/routes [hunk 3]
```diff
@@ -28,3 +28,20 @@ GET    /v1/alertsbyrule/             com.microsoft.azure.iotsolutions.devicetele
...
+GET    /v1/rules/                    com.microsoft.azure.iotsolutions.devicetelemetry.webservice.v1.controllers.RulesController.list
+
+GET    /v1/rules/:rule_id            com.microsoft.azure.iotsolutions.devicetelemetry.webservice.v1.controllers.RulesController.get(rule_id)
+GET    /v1/rules/:rule_id/           com.microsoft.azure.iotsolutions.devicetelemetry.webservice.v1.controllers.RulesController.get(rule_id)
+
+POST   /v1/rules                     com.microsoft.azure.iotsolutions.devicetelemetry.webservice.v1.controllers.RulesController.post(String = null)
+POST   /v1/rules/                    com.microsoft.azure.iotsolutions.devicetelemetry.webservice.v1.controllers.RulesController.post(String = null)
+
+POST  /v1/rules/:rule_id             com.microsoft.azure.iotsolutions.devicetelemetry.webservice.v1.controllers.RulesController.post(rule_id)
+POST  /v1/rules/:rule_id/            com.microsoft.azure.iotsolutions.devicetelemetry.webservice.v1.controllers.RulesController.post(rule_id)
```
Comment (dluc) [inline conf/routes]: the method here should be PUT

File: app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/RulesController.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+     * @return One rule.
+     */
+    public Result get(String id) {
+        return ok(toJson("TODO"));
+    }
+
+    /**
+     * @return updated rule.
+     */
+    public Result post(String id) { return ok(toJson("TODO")); }
```
Comment (dluc) [inline app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/RulesController.java]: put(...)

**Pull request merged**